nixos/modules/services/web-apps/rundeck.nix

```diff
@@ -0,0 +1,550 @@
+              "dataSource.url" =
+                if cfg.database.type == "h2" then
+                  "jdbc:h2:file:${cfg.dataDir}/data/rundeckdb;DB_CLOSE_ON_EXIT=FALSE;LOCK_TIMEOUT=10000;MODE=MYSQL"
+                else if cfg.database.type == "postgresql" then
+                  "jdbc:postgresql://${cfg.database.host}:${toString cfg.database.port}/${cfg.database.name}"
+                else
+                  "jdbc:mysql://${cfg.database.host}:${toString cfg.database.port}/${cfg.database.name}?autoReconnect=true&useSSL=false";
+              "dataSource.driverClassName" =
+                if cfg.database.type == "h2" then
+                  "org.h2.Driver"
+                else if cfg.database.type == "postgresql" then
+                  "org.postgresql.Driver"
+                else
+                  "org.mariadb.jdbc.Driver";
+              "dataSource.username" = cfg.database.username;
+              "dataSource.password" = cfg.database.password;
+            }
+            // cfg.extraConfig
+            // lib.optionalAttrs (cfg.database.type == "h2") {
+              "dataSource.dialect" = "org.hibernate.dialect.H2Dialect";
+            }
+            // lib.optionalAttrs (cfg.database.type == "mysql") {
+              "dataSource.dialect" = "org.hibernate.dialect.MariaDB103Dialect";
+            }
+            // lib.optionalAttrs cfg.ssl.enable {
+              "server.https.port" = "4443";
+              "server.ssl.keyStore" = toString cfg.ssl.keyStore;
+              "server.ssl.keyStorePassword" = cfg.ssl.keyStorePassword;
+              "server.ssl.keyPassword" = cfg.ssl.keyPassword;
+            }
+          )
+        )}
+        EOF
+
+        cat > ${cfg.configDir}/framework.properties <<EOF
+        framework.server.name = ${cfg.serverHostname}
+        framework.server.hostname = ${cfg.serverHostname}
+        framework.server.port = ${toString cfg.serverPort}
+        framework.server.url = ${
+          if cfg.serverURL != "" then
+            cfg.serverURL
+          else
+            "http${lib.optionalString cfg.ssl.enable "s"}://${cfg.serverHostname}"
+        }
+        rdeck.base = ${cfg.dataDir}
+        framework.projects.dir = ${cfg.dataDir}/projects
+        framework.etc.dir = ${cfg.configDir}
+        framework.var.dir = ${cfg.dataDir}/var
+        framework.tmp.dir = ${cfg.dataDir}/var/tmp
+        framework.logs.dir = ${cfg.dataDir}/var/logs
+        framework.libext.dir = ${cfg.dataDir}/libext
+        framework.ssh.keypath = ${cfg.dataDir}/.ssh/id_rsa
+        framework.ssh.user = ${cfg.user}
+        framework.ssh.timeout = ${toString cfg.sshTimeout}
+        rundeck.server.uuid = ${
+          if toString cfg.serverUUID == "" then "$(cat ${cfg.dataDir}/.uuid)" else toString cfg.serverUUID
+        }
+        EOF
+
+        if [ -f ${cfg.dataDir}/etc/framework.properties ]; then
+          cp -a "${cfg.configDir}/framework.properties" "${cfg.dataDir}/etc/framework.properties"
```

Comment:
We want to use tmpfiles here too and a symlink if possible

Reply:
Im not sure tmpfiles is ok in this case too.
They have some information 
Also this is a generated file from rundeck so this cp is going to replace this file.